litestar/contrib/sqlalchemy/repository/_async.py

```diff
@@ -64,7 +75,15 @@ def __init__(
         self.auto_refresh = auto_refresh
         self.auto_commit = auto_commit
         self.session = session
-        self.statement = statement if statement is not None else select(self.model_type)
+        if isinstance(statement, Select):
+            self.statement = lambda_stmt(lambda: statement)
+        elif statement is None:
+            statement = select(self.model_type)
+            self.statement = lambda_stmt(lambda: statement)
+        else:
+            self.statement = statement
+
+        self.statement = self._to_lambda_stmt(statement)
```

Comment:
Couldn't this be simplified?

```suggestion
        self.statement = self._to_lambda_stmt(statement if statement is not None else select(self.model_type))
```

Reply:
I didn't do this based on[ these guidelines here.](https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#adding-relationships-to-mapped-classes-after-declaration)

Reply:
Oh. I was not aware of that. Seems like you need to be quite careful when implementing this lambda stuff.

Reply:
We'd need a slightly different implementation of `to_lambda_stmt` because it expects `self.statement` to be set when it's called.